### PR TITLE
Windows: Check and raise for invalid icon formats. (#5755)

### LINKED ISSUE #5755: getting something error like this " %1 is not a valid Win32 application. (Error code 193) "
+++ ONLY TEXT +++ DO NOT POST IMAGES +++

## Context information (for bug reports)

* Output of `pyinstaller --version`: ```(paste here)```
* Version of Python: e.g. 3.7
* Platform: e.g GNU/Linux (distribution), Windows (language settings), OS X, conda, FreeBSD
* Did you also try this on another platform? Does it work there?

* try the latest development version, using the following command:

```shell
pip install https://github.com/pyinstaller/pyinstaller/archive/develop.zip
```

* follow *all* the instructions in our "If Things Go Wrong" Guide
  (https://github.com/pyinstaller/pyinstaller/wiki/If-Things-Go-Wrong) and

### Make sure [everything is packaged correctly](https://github.com/pyinstaller/pyinstaller/wiki/How-to-Report-Bugs#make-sure-everything-is-packaged-correctly)

  * [x] start with clean installation
  * [x] use the latest development version
  * [x] Run your frozen program **from a command window (shell)** — instead of double-clicking on it
  * [x] Package your program in **--onedir mode**
  * [x] Package **without UPX**, say: use the option `--noupx` or set `upx=False` in your .spec-file
  * [x] Repackage you application in **verbose/debug mode**. For this, pass the option `--debug` to `pyi-makespec` or `pyinstaller` or use `EXE(..., debug=1, ...)` in your .spec file.

### A minimal example program which shows the error

```
(paste text here)
“Minimal“ means: remove everything from your code which is not relevant for this bug,
esp. don't use external programs, remote requests, etc.
A very good example is https://gist.github.com/ronen/024cdae9ff2d50488438. This one helped
us reproducing and fixing a quite complex problem within approx 1 hour.
```

### Stacktrace / full error message

```
53243 INFO: Warnings written to D:\NS Word\build\main_complete\warn-main_complete.txt
53468 INFO: Graph cross-reference written to D:\NS Word\build\main_complete\xref-main_complete.html
53587 INFO: checking PYZ
53624 INFO: Building PYZ because PYZ-00.toc is non existent
53639 INFO: Building PYZ (ZlibArchive) D:\NS Word\build\main_complete\PYZ-00.pyz
57333 INFO: Building PYZ (ZlibArchive) D:\NS Word\build\main_complete\PYZ-00.pyz completed successfully.
57528 INFO: checking PKG
57578 INFO: Building PKG because PKG-00.toc is non existent
57591 INFO: Building PKG (CArchive) PKG-00.pkg
109557 INFO: Building PKG (CArchive) PKG-00.pkg completed successfully.
109587 INFO: Bootloader c:\users\lenovo\appdata\local\programs\python\python39\lib\site-packages\PyInstaller\bootloader\Windows-64bit\runw.exe
109590 INFO: checking EXE
109591 INFO: Building EXE because EXE-00.toc is non existent
109592 INFO: Building EXE from EXE-00.toc
109915 INFO: Copying icons from D:\NS Word\google_docs.png
Unable to load icon file D:\NS Word\google_docs.png
    %1 is not a valid Win32 application. (Error code 193)
```

Please also see <https://github.com/pyinstaller/pyinstaller/wiki/How-to-Report-Bugs>
for more about what would use to solve the issue.

### PR DESCRIPTION
**Pull request by bwoodsend**
Closes #5755. 

We used to let the Windows API's error propagate whenever an invalid/unsupported icon file is given but, because that API is really looking for an executable to copy an icon from, it's very obscure and insinuates than it wants to run the image as an executable.

So provide a message that says clearly *png files are not allowed - use ico instead*.

### CODE REVIEW

File: PyInstaller/utils/win32/icon.py [hunk 1]
```diff
@@ -224,6 +224,18 @@ def splitter(s):
     else:
         logger.info("Copying icons from %s", srcpath)
 
+    # Bail out quickly if the input is invalid. Letting images in the wrong
+    # format be passed to Window's API gives very cryptic error messages as
+    # it's generally unclear why PyInstaller would treat an image file as an
+    # executable.
+    if srcext != ".exe":
+        raise SystemExit(
```
Comment (BoboTiG) [inline PyInstaller/utils/win32/icon.py:232]: A `ValueError` would be more suited, maybe?

Comment (bwoodsend) [reply PyInstaller/utils/win32/icon.py:232]: I was mirroring the behavior of [L249](https://github.com/pyinstaller/pyinstaller/blob/8825887df336ae7ee8b56a25c87450b8333c0aa4/PyInstaller/utils/win32/icon.py#L249) which also calls `SystemExit`. But I would generally prefer it be a `ValueError`.

Comment (bwoodsend) [reply PyInstaller/utils/win32/icon.py:232]: Actually, sod consistency. I'm changing it.

Comment (bwoodsend) [reply PyInstaller/utils/win32/icon.py]: I see that function chooses to raise a `ValueError` if it gets more than one `.exe` so consistency is already ruined :stuck_out_tongue: ...

**Review by BoboTiG — APPROVED** [review]

File: PyInstaller/utils/win32/icon.py (hunk 1 above)
Comment (BoboTiG) [reply PyInstaller/utils/win32/icon.py]: Ah yes, L249 is brutal :)

**Review by rokm — APPROVED** [review]

**Comment by bwoodsend** [comment]
Oh wow! I believe that [this](https://github.com/pyinstaller/pyinstaller/pull/5758/checks?check_run_id=2392089646) is the first linter error I have ever seen which pointed to a genuine mistake.

**Comment by its-shubham2323** [comment]
> Oh wow! I believe that [this](https://github.com/pyinstaller/pyinstaller/pull/5758/checks?check_run_id=2392089646) is the first linter error I have ever seen which pointed to a genuine mistake.

My pleasure to help you noticing this error. 🙂

**Pull request merged**